src/data/history.ts

```diff
@@ -417,28 +429,35 @@ export const computeHistory = (
       return;
     }
 
+    const domain = computeDomain(entityId);
+
     const currentState =
       entityId in hass.states ? hass.states[entityId] : undefined;
-    const stateWithUnitorStateClass =
+    const numericStateFromHistory =
       !currentState &&
-      stateInfo.find((state) => state.a && attributesHaveUnits(state.a));
+      stateInfo.find((state) => state.a && isNumericFromAttributes(state.a));
 
     let unit: string | undefined;
 
-    if (currentState && stateUsesUnits(currentState)) {
-      unit = currentState.attributes.unit_of_measurement || " ";
-    } else if (stateWithUnitorStateClass) {
-      unit = stateWithUnitorStateClass.a.unit_of_measurement || " ";
+    const isNumeric =
+      isNumericFromDomain(domain) ||
+      (currentState != null &&
+        isNumericFromAttributes(currentState.attributes)) ||
+      (currentState != null &&
+        domain === "sensor" &&
+        isNumericSensorEntity(currentState, sensorNumericalDeviceClasses));
+
+    if (isNumeric) {
```

Comment:
I guess in that case for now it will be correct, because it will not have a UOM and will use the correct `BLANK_UNIT`, but I think that is pretty error prone...?

I would prefer:
```
      unit = currentState?.attributes.unit_of_measurement || numericStateFromHistory?.a.unit_of_measurement || BLANK_UNIT;
```